### PR TITLE
100 consider highest pure runout level

### PR DESCRIPTION
**Pull request by old2rng**
#100

### CODE REVIEW

**Comment by old2rng** [comment]
I think I made a mistake. It should be merged to main... Sorry!

File: tests/materialdata/woehler/test_analyzer.py [hunk 1]
```diff
@@ -595,3 +595,43 @@ def test_max_likelihood_one_mixed_horizon():
         wc = ml.analyze().sort_index()
     bic = ml.bayesian_information_criterion()
     pd.testing.assert_series_equal(wc, expected, rtol=1e-1)
+
+
+def test_irrelevant_runouts_dropped():
+    fd = woehler.determine_fractures(data_pure_runout_horizon_and_mixed_horizons, 1e7).fatigue_data
+    num_data_before = len(fd._obj)
+    num_tests_lowest_pure_runout_load_level = len(fd.load[fd.load==fd.pure_runout_loads.min()])
```
Comment (johannes-mueller) [inline tests/materialdata/woehler/test_analyzer.py:603]: Please hard code the values of the two `num_` variables in order to make tests simpler and avoid to make the same thinko in the tests as in the production code.

File: tests/materialdata/woehler/test_analyzer.py [hunk 2]
```diff
@@ -595,3 +595,43 @@ def test_max_likelihood_one_mixed_horizon():
...
+        'TN': 6.779709,
+        'failure_probability': 0.5
+    }).sort_index()
+
+    fd = woehler.determine_fractures(data_pure_runout_horizon_and_mixed_horizons, 1e7).fatigue_data
+    num_data_before = len(fd._obj)
+    wc = woehler.Probit(fd).analyze().sort_index()
+    num_data_after = len(fd._obj)
+    pd.testing.assert_series_equal(wc, expected, rtol=1e-1)
+    assert num_data_before == num_data_after
```
Comment (johannes-mueller) [inline tests/materialdata/woehler/test_analyzer.py:636]: Please add the same test with `new_row` from the test before.

File: tests/materialdata/woehler/test_analyzer.py [hunk 3]
```diff
@@ -595,3 +595,43 @@ def test_max_likelihood_one_mixed_horizon():
...
+    fd = fd.irrelevant_runouts_dropped()
+    num_data_after = len(fd._obj)
+    assert num_data_before-num_tests_lowest_pure_runout_load_level == num_data_after
+
+
+def test_irrelevant_runouts_dropped_no_change():
+    data_pure_runout_horizon_and_mixed_horizons
+    data_extend = data_pure_runout_horizon_and_mixed_horizons.copy(deep=True)
+    new_row = {'load': 2.75e+02, 'cycles': 1.00e+06}
+    data_extend = pd.concat([data_extend, pd.DataFrame([new_row])])#, ignore_index=True)
```
Comment (johannes-mueller) [inline tests/materialdata/woehler/test_analyzer.py:613]: Please remove the comment at the end of the line

File: src/pylife/materialdata/woehler/fatigue_data.py [hunk 4]
```diff
@@ -189,6 +205,10 @@ def _calc_finite_zone_manual(self, limit):
         self._infinite_zone = self._obj[self._obj.load <= limit]
 
 
+
+
+
+
```
Comment (johannes-mueller) [inline src/pylife/materialdata/woehler/fatigue_data.py:211]: Please remove these three empty lines.

File: src/pylife/materialdata/woehler/fatigue_data.py [hunk 5]
```diff
@@ -160,6 +164,18 @@ def set_fatigue_limit(self, fatigue_limit):
 
         return self
 
+    def irrelevant_runouts_dropped(self):
+        '''Checks if data has several pure runout load levels and drops all except max pure runout level
```
Comment (johannes-mueller) [inline src/pylife/materialdata/woehler/fatigue_data.py:168]: Consider changing to "Make a copy of the instance with irrelevant pure runout levels dropped."

**Review by johannes-mueller — CHANGES_REQUESTED** [review]
Thanks for tackling this. Please consider the comments in the code

File: tests/materialdata/woehler/test_analyzer.py (hunk 1 above)
Comment (old2rng) [reply tests/materialdata/woehler/test_analyzer.py:603]: Hey Johannes,
thanks for the feed back. I added the request.
BR, Dominic

File: tests/materialdata/woehler/test_analyzer.py (hunk 2 above)
Comment (old2rng) [reply tests/materialdata/woehler/test_analyzer.py:636]: Hey Johannes,
thanks for the feed back. I added the new test.
BR, Dominic

File: tests/materialdata/woehler/test_analyzer.py (hunk 3 above)
Comment (old2rng) [reply tests/materialdata/woehler/test_analyzer.py:613]: Done :-)

File: src/pylife/materialdata/woehler/fatigue_data.py (hunk 5 above)
Comment (old2rng) [reply src/pylife/materialdata/woehler/fatigue_data.py:168]: Yes, it sounds much better! THX

**Review by johannes-mueller — APPROVED** [review]
Thanks. Rebased to fix the DCO check.

**Pull request merged**